### PR TITLE
Upgrade to latest Scrapy with Python3 support. Upgrade/add dependencies:

### PR DESCRIPTION
**Pull request by berdario**
###### Things done
- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
  - [ ] NixOS
  - [ ] OS X
  - [X] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

parsel, pydispatcher, twisted

### CODE REVIEW

**Comment by FRidh** [comment]
please have a commit for each package you change

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -17041,6 +17041,29 @@ in modules // {
     };
   };
 
+  parsel = pkgs.python35Packages.buildPythonPackage rec {
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: `pkgs.python35Packages.` should not be here

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -17041,6 +17041,29 @@ in modules // {
...
+    version = "1.0.3";
+
+    src = pkgs.fetchurl {
+      url = "mirror://pypi/p/parsel/${name}.tar.gz";
+      sha256 = "9c12c370feda864c2f541cecce9bfb3a2a682c6c59c097a852e7b040dc6b8431";
+    };
+
+    buildInputs = with pkgs.python35Packages; [ pytest ];
+    propagatedBuildInputs = with pkgs.python35Packages; [ six w3lib lxml cssselect ];
+
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: also not here

Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: if the package should only be available with `python35`, then you can use `disabled = !isPy35;`

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -18752,13 +18775,18 @@ in modules // {
...
   pydispatcher = buildPythonPackage (rec {
     version = "2.0.5";
-    disabled = isPy35;
     name = "pydispatcher-${version}";
     src = pkgs.fetchurl {
       url = "mirror://pypi/P/PyDispatcher/PyDispatcher-${version}.tar.gz";
       sha256 = "1bswbmhlbqdxlgbxlb6xrlm4k253sg8nvpl1whgsys8p3fg0cw2m";
     };
 
+    buildInputs = with pkgs.python35Packages; [ pytest ];
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: `pkgs.python35Packages.`

**Review by FRidh — CHANGES_REQUESTED** [review]

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (berdario) [inline pkgs/top-level/python-packages.nix]: yeah, I forgot to make some changes... I tested this only with Python3.5, but ideally should work with all recent versions

I was hoping to catch failures with other Pythons with the CI

**Comment by berdario** [comment]
I've applied the changes

**Comment by FRidh** [comment]
Thanks.

In the future, please use the full attribute name for the packages, so `pythonPackages.parsel`.

**Comment by FRidh** [comment]
Regarding https://github.com/NixOS/nixpkgs/pull/16864#issuecomment-247878531

No, it's quite common that the log size gets too big. You can ignore that result.

**Pull request merged**